src/disjoint_set.jl

```diff
@@ -84,17 +87,48 @@ function root_union!(s::IntDisjointSets, x::Integer, y::Integer)
     x
 end
 
+function root_union!(s::IntDisjointSets{<:Int}, x::Integer, y::Integer)
+    parents = s.parents
+    rks = s.ranks
+    @inbounds xrank = rks[x]
+    @inbounds yrank = rks[y]
+
+    if xrank < yrank
+        x, y = y, x
+    elseif xrank == yrank
+        rks[x] += 1
+    end
+    @inbounds parents[y] = x
+    @inbounds s.sizes[x] += s.sizes[y]
+    @inbounds s.ngroups -= 1
+    x
+end
+
 # make a new subset with an automatically chosen new element x
 # returns the new element
 #
-function push!(s::IntDisjointSets)
+function push!(s::IntDisjointSets{<:Void})
+    x = length(s) + 1
+    push!(s.parents, x)
+    push!(s.ranks, 0)
+    s.ngroups += 1
+    return x
+end
+
+function push!(s::IntDisjointSets{<:Int})
     x = length(s) + 1
     push!(s.parents, x)
     push!(s.ranks, 0)
+    push!(s.sizes, 1)
     s.ngroups += 1
     return x
 end
 
+# get size of set containing element x
+#
+get_size(s::IntDisjointSets{<:Int}, x::Integer) = s.sizes[find_root(s, x)]
+get_size(s::IntDisjointSets{<:Void}, x::Integer) = error("DisjointSets wasn't initialized with store_size = true")
```

Comment:
Can you please change this to throw an `ErrorException`, rather than calling `error`?  We should be moving away from `error` in this package.